### ISSUE TITLE
Dealing with errors in production should be easy, documented, and strongly recommended

### ISSUE THREAD

**Issue by anglinb** [probot/friction#2]
**tl;dr** GitHub Apps, by their nature as bots that interact with the GitHub API, have no good way to report errors back to users. We should encourage exception tracking in production and provide strategies and/or features to deal with unrecoverable/unexpected errors. 

--- 

### Where do most developers learn to setup error tracking?
As a senior in college, one of the topics that I've never heard once in my curriculum is exception tracking, and honestly most topics surrounding deployment are completely skipped as well. I've been fortunate enough to learn about exception tracking through internships & projects and I turned a corner as a developer when I was able to feel confident about the experience the end user was receiving. 

**Question:** Do _most_ developers understand error tracking where saying provide a `SENTRY_URL` is all it takes or should we take time to educate people about production tracking through documentation and opt-out based mechanisms?

My intuition says we can assume most people wouldn't be bothered by a bit of refresher and an easy mechanism to wire up error tracking, but I've been wrong before so I would love feedback about this hypothesis and question above. 

### Sentry
I spent some time installing raven in my plugin codebase only to go back and realize Raven was already setup by Probot itself 🙃 Total derp on my part b/c I def saw [this code](https://github.com/probot/probot/blob/master/index.js#L48-L55) at some point but documenting this in [`docs/deployment.md`](https://github.com/probot/probot/blob/master/docs/deployment.md) and [`robot.js` jsdoc](https://github.com/probot/probot/blob/master/lib/robot.js#L242) would be helpful for future developers. 👍 

 I think understanding the significance of error tracking in production and sane ways to set it up are _especially_ crucial for developing a GitHub App w/ a solid user experience b/c if the bot doesn't respond there is no error delivered to the end user, which leaves end users frustrated and confused (they have told me 😜 ) . Discussed more below 👇 

### End user error alerting
[Edit] Moved this section to #1 b/c it had already been opened addressing this specific need 🙃 

**Actionable Items:**

- [ ] Document Raven usage in [`docs/deployment.md`](https://github.com/probot/probot/blob/master/docs/deployment.md) and [`robot.js` jsdoc](https://github.com/probot/probot/blob/master/lib/robot.js#L242) 📄 
- [ ] Provide a brief explanation about error tracking in prod in [`docs/deployment.md`](https://github.com/probot/probot/blob/master/docs/deployment.md)  📄
